src/main/java/org/generationcp/middleware/service/impl/study/StudyGermplasmListServiceImpl.java

```diff
@@ -24,32 +33,81 @@ public StudyGermplasmListServiceImpl(final HibernateSessionProvider sessionProvi
-			studyGermplasmDto.setEntryType(listDataProject.getList().getType());
-			studyGermplasmDto.setCheckType(listDataProject.getCheckType());
-			studyGermplasmDto.setStockIds(listDataProject.getStockIDs());
-			studyGermplasmDto.setGroupId(listDataProject.getGroupId());
+			studyGermplasmDto.setCross(this.findStockPropValue(TermId.CROSS.getId(), stockModel.getProperties()));
+			studyGermplasmDto.setDesignation(stockModel.getName());
+			studyGermplasmDto.setEntryCode(stockModel.getValue());
+			studyGermplasmDto.setEntryNumber(Integer.valueOf(stockModel.getUniqueName()));
+			studyGermplasmDto.setGermplasmId(stockModel.getGermplasm().getGid());
+			studyGermplasmDto.setPosition(String.valueOf(++index));
+			studyGermplasmDto.setSeedSource(this.findStockPropValue(TermId.SEED_SOURCE.getId(), stockModel.getProperties()));
+			studyGermplasmDto.setCheckType(Integer.valueOf(this.findStockPropValue(TermId.ENTRY_TYPE.getId(), stockModel.getProperties())));
+			studyGermplasmDto.setStockIds(stockIdsMap.getOrDefault(stockModel.getGermplasm().getGid(), ""));
+			studyGermplasmDto.setGroupId(stockModel.getGermplasm().getMgid());
 			studyGermplasmDtos.add(studyGermplasmDto);
 		}
 		return studyGermplasmDtos;
 	}
 
 	@Override
 	public List<StudyGermplasmDto> getGermplasmListFromPlots(final int studyBusinessIdentifier, final Set<Integer> plotNos) {
-		return daoFactory.getStockDao().getStudyGermplasmDtoList(studyBusinessIdentifier, plotNos);
+		return this.daoFactory.getStockDao().getStudyGermplasmDtoList(studyBusinessIdentifier, plotNos);
+	}
+
+	@Override
+	public long countStudyGermplasmList(final int studyId) {
+		return this.daoFactory.getStockDao().countStocksForStudy(studyId);
+	}
+
+	@Override
+	public void deleteStudyGermplasmList(final int studyId) {
+		this.daoFactory.getStockDao().deleteStocksForStudy(studyId);
+	}
+
+	@Override
+	public void saveStudyGermplasmList(final List<StockModel> stockModelList) {
+		for (final StockModel stockModel : stockModelList) {
+			this.daoFactory.getStockDao().saveOrUpdate(stockModel);
+		}
+	}
+
+	@Override
+	public long countStudyGermplasmByEntryTypeIds(final int studyId, final List<String> systemDefinedEntryTypeIds) {
+		return this.daoFactory.getStockDao().countStocksByStudyAndEntryTypeIds(studyId, systemDefinedEntryTypeIds);
+	}
+
+	@Override
+	public Map<Integer, String> getInventoryStockIdMap(final List<StudyGermplasmDto> studyGermplasmDtoList) {
+
+		final List<Integer> gids = new ArrayList<>();
+		if (studyGermplasmDtoList != null && !studyGermplasmDtoList.isEmpty()) {
+			for (final StudyGermplasmDto studyGermplasmDto : studyGermplasmDtoList) {
+				gids.add(studyGermplasmDto.getGermplasmId());
+			}
+		}
+
+		return this.inventoryDataManager.retrieveStockIds(gids);
+
+	}
+
+	private String findStockPropValue(final int termId, final Set<StockProperty> properties) {
+		if (properties != null) {
+			for (final StockProperty property : properties) {
+				if (termId == property.getTypeId()) {
```

Comment:
this comparisson could cause issues that are very difficult to find out. I think it will be better to make termId an Integer and compare using equals